tests/integration-tests-topologies/src/main/java/org/apache/pulsar/tests/topologies/PulsarCluster.java

```diff
@@ -193,24 +208,25 @@ public void start() throws Exception {
             String name = "pulsar-" + serviceName + "-" + i;
             T container = containerCreator.apply(name);
             containers.put(name, container);
-            startFutures.add(CompletableFuture.runAsync(() -> container.start()));
         }
-        CompletableFuture.allOf(startFutures.toArray(new CompletableFuture[startFutures.size()])).join();
-        log.info("Successfully started {} {} containers", numContainers, serviceName);
         return containers;
     }
 
     public void stop() {
-        Stream.of(proxyContainer, csContainer, zkContainer).parallel().forEach(GenericContainer::stop);
-        workerContainers.values().parallelStream().forEach(GenericContainer::stop);
-        brokerContainers.values().parallelStream().forEach(GenericContainer::stop);
-        bookieContainers.values().parallelStream().forEach(GenericContainer::stop);
-        if (null != spec.externalServices()) {
-            spec.externalServices().values()
-                .parallelStream()
-                .forEach(GenericContainer::stop);
+
+        Stream<GenericContainer> containers = Streams.concat(
+                workerContainers.values().stream(),
+                brokerContainers.values().stream(),
+                bookieContainers.values().stream(),
+                Stream.of(proxyContainer, csContainer, zkContainer)
+        );
+
+        if (spec.externalServices() != null) {
+            containers = Streams.concat(containers, spec.externalServices().values().stream());
         }
 
+        containers.forEach(GenericContainer::stop);
```

Comment:
isn't this changed from parallelStream to sequential?

Reply:
it's parallel now